src/openms/include/OpenMS/QC/Suitability.h

```diff
@@ -0,0 +1,173 @@
+
+#pragma once
+
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+
+#include <cfloat>
+#include <map>
+#include <vector>
+
+namespace OpenMS
+{
+  class PeptideIdentification;
+  class PeptideHit;
+
+  /**
+  * @brief This class serves as the library representation @ref TOPP_DatabaseSuitability
+  *
+  * This class holds the functionality of calculating the database suitability.
+  * This can only be done if a combined deNovo+database identification search was performed.
+  * Currently only Comet search is supported.
+  *
+  * Allows for multiple usage, because results are stored internally and can be returned
+  * using getResults().
+  *
+  */
+  class OPENMS_DLLAPI Suitability:
+    public DefaultParamHandler
+  {
+  public:
+    /// struct to store results
+    struct SuitabilityData
+    {
+      Size num_top_novo = 0;
+      Size num_top_db = 0;
+      Size num_re_ranked = 0;
+      Size num_interest = 0;
+      double cut_off = DBL_MAX;
+      double suitability = 0;
+    };
+
+    /// Constructor
+    /// Settings are initialized with their default values:
+    /// no_re_rank = false, novo_fract = 1, FDR = 0.01
+    Suitability();
+
+    /// Destructor
+    ~Suitability() = default;
+
+    /**
+    * @brief Computes suitability of a database used to search a mzML
+    *
+    * Counts top deNovo and top database hits. The ratio of db hits vs
+    * all hits yields the suitability.
+    * To re-rank cases, where a de novo peptide scores just higher than
+    * the database peptide, a decoy cut-off is calculated. This functionality
+    * can be turned off. This will result in an underestimated suitability,
+    * but it can solve problems like different search engines or to few decoy hits.
+    * 
+    * Parameters can be set using the functionality of DefaultParamHandler.
+    * Parameters are:
+    *                no_re_rank - re-ranking can be turned off with this
+    *                novo_fract - percent of deNovo peptides to capture in re-ranking
+    *                FDR        - q-value that should be filtered for
+    *                             Preliminary tests have shown that database suitability
+    *                             is rather stable across common FDR thresholds from 0 - 5 %
+    *
+    * Since q-values need to be calculated the identifications are taken by copy.
+    *
+    * Result is appended to the result member. This allows for multiple usage.
+    *
+    * @param pep_ids      vector containing pepIDs coming from a deNovo+database 
+    *                     identification search without FDR (currently only Comet-support)
+    * @throws             MissingInformation if no target/decoy annotation is found
+    * @throws             MissingInformation if no xcorr is found
+    * @throws             Precondition if a q-value is found in the input
+    */
+    void compute(std::vector<PeptideIdentification> pep_ids);
+
+    /// return results
```

Comment:
can you document the output a bit more clear, how am I supposed to use this? What do I do with these results, what does the `suitability` parameter tell me? How do I now know whether the DB is suitable for the file or not? do I look at all entries in the vector?